jHipster 6.0.0: DTO mapper doesn't ignore two collection helper methods

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

I created a jHipster 6.0.0 / Java 11 project with a data model and DTOs. After fixing the wrong fake data ([#9721](https://github.com/jhipster/generator-jhipster/issues/9721)), Mapstruct warned about two unmapped target properties. Both were collection helper methods in the entities.

<!-- Explain the bug, if an error is being thrown a stack trace helps -->

Both entities have a `Set` of another entity. jHipster generated `addX` and `removeX` helper methods, without excluding either in the generated mapper interface. Mapstruct then warned about the `removeX` property not being mapped - but not about the `addX` one:

````
> Task :compileJava
/Users/karsten/workspaces/me/services/jump-start-project/newsletter/jhipster-001/jhipster/jhipster-6.0.0-java-11-test/src/main/java/com/betterprojectsfaster/jhipster/test/jhipster6java11/service/mapper/ShoppingOrderMapper.java:21: warning: Unmapped target property: "removeOrders".
    ShoppingOrder toEntity(ShoppingOrderDTO shoppingOrderDTO);
                  ^
/Users/karsten/workspaces/me/services/jump-start-project/newsletter/jhipster-001/jhipster/jhipster-6.0.0-java-11-test/src/main/java/com/betterprojectsfaster/jhipster/test/jhipster6java11/service/mapper/ProductMapper.java:16: warning: Unmapped target property: "removeOrder".
    Product toEntity(ProductDTO productDTO);
            ^
Note: Hibernate JPA 2 Static-Metamodel Generator 5.3.9.Final
2 warnings
````

##### **Motivation for or Use Case**

The generated mapper code should go through without a warning.

##### **Reproduce the error**

<!-- For bug reports, an unambiguous set of steps to reproduce the error -->

- Start a jHipster 6.0.0 / Java 11 project by importing [my JDL file](https://github.com/ksilz/jhipster-6-java-11-test/blob/master/simple-shop.jdl).
- Because of the wrong fake data ([#9721](https://github.com/jhipster/generator-jhipster/issues/9721)), you need to replace these three files in `src/main/resources/config/liquibase/data`with fixed versions from my project: [`product.csv`](https://github.com/ksilz/jhipster-6-java-11-test/blob/fake-data-fixed/src/main/resources/config/liquibase/data/product.csv), [`shipment.csv`](https://github.com/ksilz/jhipster-6-java-11-test/blob/fake-data-fixed/src/main/resources/config/liquibase/data/shipment.csv) and [`shopping_order.csv`](https://github.com/ksilz/jhipster-6-java-11-test/blob/fake-data-fixed/src/main/resources/config/liquibase/data/shopping_order.csv).
- Run `./gradlew` and you will see the warnings.

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->
- [#9721](https://github.com/jhipster/generator-jhipster/issues/9721) creates fake wrong data for my JDL

##### **Suggest a Fix**

I added a `@Mapping(target = "removeOrder", ignore = true)` and `@Mapping(target = "removeOrders", ignore = true)` annotations in the respective mapper interfaces. This fixed it. See [lines 16 & 22 in this commit](https://github.com/ksilz/jhipster-6-java-11-test/commit/57bc4e3a839ec52f260d8fd86f4c9ba85d9496e2?diff=unified) - sorry for all the other changes that obscur the view.

<!-- If you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

##### **JHipster Version(s)**

6.0.0, Java 11

<!--
Which version of JHipster are you using, is it a regression?
-->

##### **JHipster configuration**

<!--
To provide all information we need, you should run `jhipster info` in the project root folder, and
copy/paste the result here.
The `.yo-rc.json` file generated in the root folder is mandatory for bug reports. This will help us to replicate the scenario.
You should remove any sensitive information like the rememberMe key or the jwtSecretKey key.
-->

INFO! Using JHipster version installed locally in current project's node_modules
INFO! Executing jhipster:info
INFO! Options: from-cli: true
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
j-hipster-6-java-11-test@0.0.0 /Users/karsten/workspaces/me/services/jump-start-project/newsletter/jhipster-001/jhipster/jhipster-6.0.0-java-11-test
└── generator-jhipster@6.0.0 

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "enableHibernateCache": true,
    "enableSwaggerCodegen": false,
    "enableTranslation": true,
    "jhiPrefix": "bpf",
    "languages": ["en", "de"],
    "messageBroker": false,
    "nativeLanguage": "en",
    "packageName": "com.betterprojectsfaster.jhipster.test.jhipster6java11",
    "packageFolder": "com/betterprojectsfaster/jhipster/test/jhipster6java11",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "serviceDiscoveryType": false,
    "skipClient": false,
    "skipServer": false,
    "testFrameworks": ["protractor", "cucumber"],
    "websocket": "spring-websocket",
    "baseName": "jHipster6Java11Test",
    "applicationType": "monolith",
    "authenticationType": "jwt",
    "buildTool": "gradle",
    "cacheProvider": "ehcache",
    "clientFramework": "angularX",
    "useSass": true,
    "clientPackageManager": "npm",
    "jhipsterVersion": "6.0.0",
    "skipUserManagement": false,
    "serverPort": "8080",
    "jwtSecretKey": "bXktc2VjcmV0LXRva2VuLXRvLWNoYW5nZS1pbi1wcm9kdWN0aW9uLWFuZC10by1rZWVwLWluLWEtc2VjdXJlLXBsYWNl",
    "clientTheme": "none",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": []
  },
  "entities": ["Product", "Address", "ShoppingOrder", "ProductOrder", "Shipment"]
}

</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Product {
  name String required unique minlength(2) maxlength(90),
  price Float required min(0),
  description TextBlob required,
  picture ImageBlob required,
  specification AnyBlob,
  category ProductCategory,
  inventory Integer required min(0)
}
entity Address {
  addressLine1 String required minlength(2) maxlength(80),
  addressLine2 String minlength(2) maxlength(80),
  city String minlength(2) maxlength(80),
  postalCode String minlength(5) maxlength(5)
}
entity ShoppingOrder {
  name String required unique minlength(2) maxlength(90),
  totalAmount Float min(0),
  ordered ZonedDateTime
}
entity ProductOrder {
  amount Integer required min(0) max(5)
}
entity Shipment {
  shippedAt ZonedDateTime required
}

enum ProductCategory {
  Laptop,
  Desktop,
  Phone,
  Tablet,
  Accessory
}

relationship OneToOne {
  Shipment{order(name) required} to ShoppingOrder{shipment required}
}
relationship OneToMany {
  Product{order required} to ProductOrder{product(name) required},
  ShoppingOrder{orders required} to ProductOrder{overallOrder(name)}
}
relationship ManyToOne {
  Address{user(login) required} to User,
  ShoppingOrder{buyer(login) required} to User,
  ProductOrder{buyer(login) required} to User,
  Shipment{shippedBy(login) required} to User
}

dto Product, Address, ShoppingOrder, ProductOrder, Shipment with mapstruct
service Product, Address, ShoppingOrder, ProductOrder, Shipment with serviceClass

</pre>
</details>


##### **Environment and Tools**

openjdk version "1.8.0_202"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_202-b08)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.202-b08, mixed mode)

git version 2.21.0

node: v10.15.3

npm: 6.9.0

Docker version 18.09.2, build 6247962

docker-compose version 1.23.2, build 1110ad01

INFO! Congratulations, JHipster execution is complete!

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

- [`Product.java`](https://github.com/ksilz/jhipster-6-java-11-test/blob/master/src/main/java/com/betterprojectsfaster/jhipster/test/jhipster6java11/domain/Product.java), [`ProductMapper.java`](https://github.com/ksilz/jhipster-6-java-11-test/blob/master/src/main/java/com/betterprojectsfaster/jhipster/test/jhipster6java11/service/mapper/ProductMapper.java)
- [`ShoppingOrder.java`](https://github.com/ksilz/jhipster-6-java-11-test/blob/master/src/main/java/com/betterprojectsfaster/jhipster/test/jhipster6java11/domain/ShoppingOrder.java), [`ShoppingOrderMapper.java`](https://github.com/ksilz/jhipster-6-java-11-test/blob/master/src/main/java/com/betterprojectsfaster/jhipster/test/jhipster6java11/service/mapper/ShoppingOrderMapper.java)

<!--
If the error is during an entity creation or associated with a specific entity.
If you are using JDL, please share that configuration as well.
-->

##### **Browsers and Operating System**

All browsers.

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

-   [x] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/entity-server/templates/src/main/java/package/service/mapper/EntityMapper.java.ejs
+++ b/generators/entity-server/templates/src/main/java/package/service/mapper/EntityMapper.java.ejs
@@ -64,6 +64,7 @@ for (idx in relationships) {
     const relationshipName = relationships[idx].relationshipName;
     // hasOAuthUser = (relationshipName === 'user' && authenticationType === 'oauth2');
     const relationshipNamePlural = relationships[idx].relationshipNamePlural;
+    const relationshipNameCapitalized = relationships[idx].relationshipNameCapitalized;
     const ownerSide = relationships[idx].ownerSide;
     if (relationshipType === 'many-to-one' || (relationshipType === 'one-to-one' && ownerSide === true)) {
         renMapAnotDto = true;
@@ -75,8 +76,11 @@ _%>
     @Mapping(target = "<%= relationshipNamePlural %>", ignore = true)
             <%_ } else if (relationshipType === 'one-to-one' && ownerSide === false) { renMapAnotDto = true; _%>
     @Mapping(target = "<%= relationshipName %>", ignore = true)
-            <%_ } _%>
-        <%_ } _%>
+            <%_ }
+    if ((relationshipType === 'one-to-many' || relationshipType === 'many-to-many') && fluentMethods) {_%>
+    @Mapping(target = "remove<%= relationshipNameCapitalized %>", ignore = true)
+    <%_ } _%>
+    <%_ } _%>
     <%_ if (renMapAnotDto === true) { _%>
     <%= asEntity(entityClass) %> toEntity(<%= asDto(entityClass)%> <%= asDto(entityInstance) %>);
     <%_ } _%>

